tests/remote-inbox-notifications/transformer-service.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * TransformerService tests.
+ *
+ * @package WooCommerce\Admin\Tests\RemoteInboxNotifications
+ */
+
+use Automattic\WooCommerce\Admin\RemoteInboxNotifications\Transformers\ArrayKeys;
+use Automattic\WooCommerce\Admin\RemoteInboxNotifications\TransformerService;
+
+/**
+ * class WC_Tests_RemoteInboxNotifications_TransformerService
+ */
+class WC_Tests_RemoteInboxNotifications_TransformerService extends WC_Unit_Test_Case {
+	/**
+	 * Test it creates a transformer with snake case 'use' value
+	 */
+	public function test_it_creates_a_transformer_with_snake_case_use_value() {
+		$array_keys = TransformerService::create_transformer( 'array_keys' );
+		$this->assertInstanceOf( ArrayKeys::class, $array_keys );
+	}
+
+	/**
+	 * Test it returns null when a transformer is not found.
+	 */
+	public function test_it_returns_null_when_transformer_is_not_found() {
+		$transformer = TransformerService::create_transformer( 'i_do_not_exist' );
+		$this->assertNull( $transformer );
+	}
+
+	/**
+	 * @expectedException InvalidArgumentException
+	 * @expectedExceptionMessage Missing required config value: use
+	 */
+	public function test_it_throw_exception_when_transformer_config_is_missing_use() {
+		TransformerService::apply( array( 'value' ), array( new stdClass() ) );
+	}
+
+	/**
+	 * @expectedException InvalidArgumentException
+	 * @expectedExceptionMessage Unable to find a transformer by name: i_do_not_exist
+	 */
+	public function test_it_throws_exception_when_transformer_is_not_found() {
+		$transformer = $this->transformer_config( 'i_do_not_exist' );
+		TransformerService::apply( array( 'value' ), array( $transformer ) );
+	}
+
+	/**
+	 * Given two transformers
+	 * When the second transformer returns null
+	 * Then the value transformed by the first transformer should be returned.
+	 */
+	public function test_it_returns_previously_transformed_value_when_transformer_returns_null() {
```

Comment:
loving these tests

Reply:
Thank you! I like this approach too. It helps me to organize my tests before writing them :)